### PR TITLE
Adding FractionScan to puma

### PR DESCRIPTION
**Pull request by afroch**
## Summary

This pull request introduces the following changes

* `Adding FractionScan` and `FractionScanPlot`
* Add unit tests for the new cases

Relates to the following issues

* #1 
*

## Conformity
- [x] [Changelog entry](https://github.com/umami-hep/puma/blob/main/changelog.md)
- [x] [Documentation](https://umami-hep.github.io/puma/)

### CODE REVIEW

**Comment by manuguth** [comment]
can you add en example for this type of plot?

**Comment by afroch** [comment]
> can you add en example for this type of plot?

Is added

File: puma/fraction_scan.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    """
+    fraction_scan class storing info about the fractions.
+    """
+
+    def __init__(
+        self,
+        x_values: np.ndarray,
+        y_values: np.ndarray,
+        **kwargs,
+    ) -> None:
```
[the remark this answers is not in the document]
Comment (afroch) [reply puma/fraction_scan.py:19]: For __init__ you don't need docstrings by recommendation. Maybe thats the reason why darglint does not complain

[the remark this answers is not in the document]
Comment (afroch) [reply puma/fraction_scan.py:17]: I added it.

File: puma/fraction_scan.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+
+class FractionScanPlot(PlotBase):
+    """Fraction scan plot class"""
+
+    def __init__(
+        self,
+        logy: bool = False,
+        **kwargs,
+    ) -> None:
```
[the remark this answers is not in the document]
Comment (afroch) [reply puma/fraction_scan.py:101]: I added it.

File: puma/fraction_scan.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        if key in self.plot_objects:
+            raise KeyError(f"Duplicated key {key} already used for unique identifier.")
+
+        # Add key to histogram object
+        curve.key = key
+        logger.debug("Adding fraction scan of %s", key)
+
+        # Set alpha
+        if curve.alpha is None:
+            curve.alpha = 0.8
```
[the remark this answers is not in the document]
Comment (afroch) [reply puma/fraction_scan.py]: Is changed to 1

**Review by jobirk — APPROVED** [review]

**Pull request merged**